.github/workflows/push-subgraph.yml

```diff
@@ -33,49 +34,69 @@ on:
         description: 'subgraph name'
         required: true
         type: string
+
 jobs:
-  push-subgraph:
+  build:
     environment: development
-    runs-on: self-hosted
+    runs-on: 
+    - self-hosted
+    - ${{ (github.event.inputs.environment == 'development' && 'subgraphs-dev') || (github.event.inputs.environment == 'production' && 'subgraphs-prod') }}
     steps:
     - uses: actions/checkout@v3
-
+    
     - name: Set up Node.js 16
       uses: actions/setup-node@v3
       with:
         node-version: '16'
-    
+
+    - name: Install Yarn
+      run: |
+        curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+        echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+        sudo apt-get update && sudo apt-get install -y yarn
+
+    - name: Install graph-cli
+      run: |
+        sudo yarn global add @graphprotocol/graph-cli
+        
     - name: Run npm
+      working-directory: ${{ github.event.inputs.folder }}
       run: |
-        cd $FOLDER_NAME
+        echo $FOLDER_NAME
+        echo $SUBGRAPH_NAME
         npm i
         npm run codegen
         npm run build
         npm run deploy
```

Comment:
This is not required, because you are deploying in next stage

Reply:
Removed!